controllers/dspipeline_params.go

```diff
@@ -630,6 +632,38 @@ func (p *DSPAParams) ExtractParams(ctx context.Context, dsp *dspa.DataSciencePip
 			}
 		}
 
+		if cfg := p.APIServer.CustomKfpLauncherConfig; cfg != nil {
+			if cfg.Name != "" && cfg.DefaultPipelineRoot != "" && cfg.Key != "" {
+				if cm, err := util.GetConfigMap(ctx, cfg.Name, p.Namespace, client); err != nil {
+					// If the custom kfp-launcher configmap is not available, that is OK
+					if !apierrs.IsNotFound(err) {
+						log.Error(err, fmt.Sprintf("Encountered error when attempting to fetch ConfigMap: [%s], Error: %v", cfg.Name, err))
+						return err
+					}
+				} else {
+					// Consume all the required information.
+					configMapName := config.CustomKfpLauncherConfigMapName
+					defaultPipelineRootValue := util.GetConfigMapValue("defaultPipelineRoot", cm)
+					dataValues := util.GetConfigMapValues(cm)
+
+					for _, val := range dataValues {
+						if val != "" {
+							p.DataValue = append(p.DataValue, []byte(val))
+						}
+					}
+
+					p.APIServer.CustomKfpLauncherConfig = &dspa.KfpLauncherConfigMap{
+						Name:                configMapName,
+						DefaultPipelineRoot: defaultPipelineRootValue,
+						Key:                 string(bytes.Join(p.DataValue, []byte{})),
+					}
+					log.Info(p.APIServer.CustomKfpLauncherConfig.Key)
+				}
+			} else {
+				return fmt.Errorf("KFP launcher custom ConfigMap provided but the Name or DefaultPipelineRoot or Key fields are blank")
+			}
+		}
```

Comment:
based on my previous comment, you should just be able to read the configmap, store the `.data` as a `params` and utilize that in the template, reach out to me for more info on this